### PR TITLE
Fix a potential memory leak in crypto/provider_child.c

### PR DESCRIPTION
**Pull request by RuiliF**
Fix issue #20063.
CLA: trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/provider_child.c [hunk 1]
```diff
@@ -135,8 +135,10 @@ static int provider_create_child_cb(const OSSL_CORE_HANDLE *prov, void *cbdata)
                                        1)) == NULL)
             goto err;
 
-        if (!ossl_provider_activate(cprov, 0, 0))
+        if (!ossl_provider_activate(cprov, 0, 0)){
```
Comment (t8m) [inline crypto/provider_child.c:138]: Please add a space before `{`

**Review by t8m — CHANGES_REQUESTED** [review]

File: crypto/provider_child.c (hunk 1 above)
Comment (RuiliF) [reply crypto/provider_child.c]: Sorry about the formatting problem! I've updated it.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
I agree with CLA: trivial

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]
agreed trivial

**Comment by paulidale** [comment]
Merged, thanks for the fix.

**Pull request closed without merging**